### PR TITLE
[TN-5] Add syslog pump.

### PR DESCRIPTION
**Pull request by sedkis**
For #189 

Docs: https://github.com/TykTechnologies/tyk-docs/pull/1005

Fix [TN-5]

[TN-5]: https://leotyktest.atlassian.net/browse/TN-5

### CODE REVIEW

File: pumps/syslog.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	"fmt"
+	"github.com/TykTechnologies/logrus"
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/mitchellh/mapstructure"
+	"log/syslog"
+)
+
+type SyslogPump struct {
+	syslogConf *SyslogConf
+	Writer     *syslog.Writer
```
Comment (adelowo) [inline pumps/syslog.go]: It will be better not to export this.

Comment (adelowo) [reply pumps/syslog.go]:
```suggestion
      writer     *syslog.Writer
```

File: pumps/syslog.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
+package pumps
+
+import (
+	"fmt"
+	"github.com/TykTechnologies/logrus"
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/mitchellh/mapstructure"
+	"log/syslog"
+)
```
Comment (adelowo) [inline pumps/syslog.go]: can this be properly formatted?

File: pumps/syslog.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+func (s *SyslogPump) Init(config interface{}) error {
+	//Read configuration file
+	s.syslogConf = &SyslogConf{}
+	err := mapstructure.Decode(config, &s.syslogConf)
+
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"prefix": logPrefix,
+		}).Fatal("Failed to decode configuration: ", err)
```
Comment (matiasinsaurralde) [inline pumps/syslog.go]: Should we `return` here?

Comment (adelowo) [reply pumps/syslog.go]: We don't need to. `Fatal` will shut down the process

Comment (sedkis) [reply pumps/syslog.go]: I prefer to shut down the pump if the desired pumps aren't loading correctly

File: pumps/syslog.go (hunk 2 above)
Comment (sedkis) [reply pumps/syslog.go]: sorry how do you mean @adelowo ?

Comment (sedkis) [reply pumps/syslog.go]: ahh you mean with spaces! 
done now

**Comment by sedkis** [comment]
@buger , 

is this PR missing anything?  I can update the pump README and then I'll create a PR for the docs

**Comment by buger** [comment]
If our final goal is fluentD, why not to make it compatible?

**Comment by sedkis** [comment]
> If our final goal is fluentD, why not to make it compatible?

Here is where I'm struggling because Syslog is going to have a different implementation depending on the OS.  I'm fairly certain that FluentD users will regularly need to provide a parsing algorithm to properly insert Logs

**Comment by sedkis** [comment]
Added syslog sections to the Pump readme as well as the Tyk Docs

File: pumps/syslog.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	log.Debug("Syslog Pump active")
+	return nil
+}
+
+func initWriter(s *SyslogPump) {
+	syslogWriter, err := syslog.Dial(
+		s.syslogConf.Transport,
+		s.syslogConf.NetworkAddr,
+		syslog.Priority(s.syslogConf.LogLevel),
+		"syslog-pump")
```
Comment (letzya) [inline pumps/syslog.go]: tag could be configurable and defaulting to `logPrefix` 's value

Comment (tbuchaillot) [reply pumps/syslog.go]: Added `tag` to config options.

**Pull request merged**